src/Core/Streaming/Watermark.h

```diff
@@ -9,5 +9,48 @@ namespace Streaming
 /// The 1970-01-01 00:00:00 is Int64 value 0, earlier times are negative, so we need to use signed Int64 as watermark
 constexpr Int64 INVALID_WATERMARK = std::numeric_limits<Int64>::min();
 constexpr Int64 TIMEOUT_WATERMARK = std::numeric_limits<Int64>::max();
+
+
+enum class WatermarkStrategy
```

Comment:
```
/// WatermarkStrategy defines how proton observes watermark and stamps watermark for events when events
/// gets flowed from the source to the whole processing pipeline.
/// Not every events will be associated with a watermark in general since sometimes events are batched or 
/// sometimes we just don't want do that because it is not necessary or inefficient. 
```